CMakeLists.txt

```diff
@@ -1,55 +1,162 @@
+option(BUILD_TESTING "Enable testing for cpu-features dep" OFF)
 
-#----------------------------------------------------------------------------------------------
 
-project(redisearch)
-
-if (NOT MODULE_NAME)
-    set(MODULE_NAME "search")  # unless it was already set to "searchlite"
+#----------------------------------------------------------------------------------------------
+# Configure output paths based on build configuration
+set(MODULE_NAME "search" CACHE STRING "Module name" FORCE)
+if(NOT DEFINED COORD_TYPE)
+    set(COORD_TYPE "oss")
+endif()
+if(COORD_TYPE STREQUAL "oss")
+  set(BINDIR "${binroot}/search-community")
+elseif(COORD_TYPE STREQUAL "rlec")
+  set(BINDIR "${binroot}/search-enterprise")
+  add_compile_definitions(PRIVATE RS_CLUSTER_ENTERPRISE)
+  set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME "module-enterprise")
+else()
+  message(FATAL_ERROR "Invalid COORD_TYPE (='${COORD_TYPE}'). Should be either 'oss' or 'rlec'")
 endif()
 
 #----------------------------------------------------------------------------------------------
+project(redisearch)
 
+#----------------------------------------------------------------------------------------------
+# Configure compiler options
 setup_cc_options()
 
-# ugly hack for cpu_features::list_cpu_features coming from VecSim
-set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} ${CMAKE_LD_FLAGS}")
-
+# Sanitizer settings
 message(STATUS "SAN: ${SAN}")
-
-# check if env sanitizers are set, and set ld_flags accordingly
-if (SAN)
-    if (SAN STREQUAL "address")
+if(SAN)
+    if(SAN STREQUAL "address")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-omit-frame-pointer -fsanitize=address -fsanitize-recover=all")
         set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} -fsanitize=address")
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
-        # print the flags
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
         message(STATUS "CMAKE_C_FLAGS: ${CMAKE_C_FLAGS}")
         message(STATUS "CMAKE_LINKER_FLAGS: ${CMAKE_LINKER_FLAGS}")
         message(STATUS "CMAKE_SHARED_LINKER_FLAGS: ${CMAKE_SHARED_LINKER_FLAGS}")
     endif()
 endif()
 
-# The default signedness of char can vary by architecture and compiler.
-# Common defaults:
-# - x86/x86_64: Typically signed by default.
-# - ARM: Typically unsigned by default.
-# This can lead to different behavior when comparing or converting char to int,
-# and causes different behavior between different platforms when sorting strings.
-# To avoid this, we set the default signedness of char to unsigned.
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -funsigned-char")
+if (COV)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
+endif()
+
+# Get Git version info
```

Comment:
I wasn't sure where and why this is used, can you explain?

Reply:
for log when we load the module (we print the git sha)